packages/hub/routes/profiles.ts

```diff
@@ -0,0 +1,253 @@
+import shortUUID from 'short-uuid';
+import { Profile } from '@prisma/client';
+import { serializeErrors } from './utils/error';
+
+export default class ProfilesRoute {
+  profileValidator: ProfileValidator = inject('profile-validator', {
+    as: 'profileValidator',
+  });
+  profileSerializer = inject('profile-serializer', {
+    as: 'profileSerializer',
+  });
+  prismaManager = inject('prisma-manager', { as: 'prismaManager' });
+  reservedWords = inject('reserved-words', {
+    as: 'reservedWords',
+  });
+
+  workerClient = inject('worker-client', { as: 'workerClient' });
+
+  constructor() {
+    autoBind(this);
+  }
+
+  async get(ctx: Koa.Context) {
+    if (!ensureLoggedIn(ctx)) {
+      return;
+    }
+    let prisma = await this.prismaManager.getClient();
+    let profiles = await prisma.profile.findMany({
+      where: {
+        ownerAddress: ctx.state.userAddress,
+      },
+    });
+
+    ctx.type = 'application/vnd.api+json';
+    ctx.status = 200;
+    ctx.body = this.profileSerializer.serialize(profiles);
+  }
+
+  async post(ctx: Koa.Context) {
+    if (!ensureLoggedIn(ctx)) {
+      return;
+    }
+
+    if (!validateRequiredFields(ctx, { requiredAttributes: ['name', 'slug', 'color', 'text-color'] })) {
+      return;
+    }
+
+    if (ctx.request.body.data.attributes['name'].length > 50) {
+      ctx.status = 422;
+      ctx.body = {
+        status: '422',
+        title: 'Invalid profile name',
+        detail: 'Profile name cannot exceed 50 characters',
+      };
+      ctx.type = 'application/vnd.api+json';
+      return;
+    }
+
+    let slug = ctx.request.body.data.attributes['slug'].toLowerCase();
+    let validationResult = await this.validateSlug(slug);
+
+    if (!validationResult.slugAvailable) {
+      ctx.status = 422;
+      ctx.body = {
+        status: '422',
+        title: 'Invalid profile slug',
+        detail: validationResult.detail,
+      };
+      ctx.type = 'application/vnd.api+json';
+      return;
+    }
+
+    let prisma = await this.prismaManager.getClient();
+    let properties = {
+      id: shortUuid.uuid(),
+      name: ctx.request.body.data.attributes['name'],
+      slug,
+      color: ctx.request.body.data.attributes['color'],
+      textColor: ctx.request.body.data.attributes['text-color'],
+      profileDescription: ctx.request.body.data.attributes['profile-description'],
+      profileImageUrl: ctx.request.body.data.attributes['profile-image-url'],
+      links: ctx.request.body.data.attributes['links'],
+      ownerAddress: ctx.state.userAddress,
+    };
+
+    let profile: Profile;
+
+    profile = await prisma.profile.create({ data: { ...properties } });
+
+    await this.workerClient.addJob('persist-off-chain-merchant-info', {
+      id: profile.id,
+    });
+
+    let serialized = this.profileSerializer.serialize(profile);
+
+    ctx.status = 201;
+    ctx.body = serialized;
+    ctx.type = 'application/vnd.api+json';
+  }
+
+  async patch(ctx: Koa.Context) {
```

Comment:
Should this also have an action to update S3?

Reply:
Maybe in this case it should not be a job to update, but rather the hub calling and waiting for the S3 update.

Reply:
hmm why not use a job? We expect a delay after `POST`, why not `PATCH`? But agreed that this should trigger an update, added now, thanks.

Reply:
Because for `PATCH`, users can refresh after the action seemingly completes, and get back a profile that does not include the changes, possibly leading to confusion about whether changes actually were submitted, and if so, when they will be reflected.

Reply:
let’s revisit when we see `PATCH` in action and refine as needed 🤓